python_modules/dagster/dagster/config/type_printer.py

```diff
@@ -47,6 +47,18 @@ def _do_print(config_schema_snapshot, config_type_key, printer, with_lines=True)
         printer.append(" | ")
         _do_print(config_schema_snapshot, config_type_snap.non_scalar_type_key, printer)
         printer.append(")")
+    elif kind == ConfigTypeKind.KEYED_COLLECTION:
+        line_break_fn("{")
+        with printer.with_indent():
+            printer.append("*String: ")
+            _do_print(
+                config_schema_snapshot,
+                config_type_snap.inner_type_key,
+                printer,
+                with_lines=with_lines,
+            )
+            line_break_fn("")
+        printer.append("}")
```

Comment:
`{*String: X}`  feels a bit rough to me, though it is inline with the existing patterns. Riffing:
 `{str -> X}`
 `{String -> X}`
 `{key -> X}`

Reply:
This has been changed with the implementation of the key type field. The new format is
`{[String]: X}`
I'm not fully convinced by this format, but it does look similar to [typescript map types](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html). Otherwise we could use the arrow notation.